shared/lib/delegation/delegation.ts

```diff
@@ -245,3 +258,35 @@ export const encodeDisableDelegation = ({
 
   return concat([encodedSignature, encodedData]);
 };
+
+/**
+ * Encodes the calldata for a redeemDelegations(delegations,modes,executions) call.
+ *
+ * @param params
+ * @param params.delegations - The delegations to redeem.
+ * @param params.modes - The modes to redeem the delegations with.
+ * @param params.executions - The executions to redeem the delegations with.
+ * @returns The encoded calldata.
+ */
+export const encodeRedeemDelegations = ({
```

Comment:
This duplicates some of the logic in `app/scripts/lib/transaction/delegation.ts`.

Though I assumed that (transactions) code was temporary and we'd be using the SDK package directly going forward?

Is there a reason we're manually implementing here?

Reply:
@matthewwalsh0  Yes, the plan is to use it directly from the SDK, but we don’t have the new SDK package yet that will isolate these functions. (The Delegation team is planning to use their `utils` package as the new core package.)

Also, one of the reasons we added this new `encodeRedeemDelegations ` function was because the existing one wasn’t working for our case. Not sure if it’s an issue during the encoding, but we basically copied the `encodeRedeemDelegations ` function from the SDK, similar to what we’ve done before with other functions.

Here’s a transaction that failed when using the existing `encodeRedeemDelegations ` function:
https://sepolia.etherscan.io/tx/0xc9f2b58e78ec851bfa34b3ce59237572fb35f2dcc0cd691bde4136cee9214981